lib/main.dart

```diff
@@ -1,125 +1,145 @@
+      appBar: AppBar(title: Text(widget.title), centerTitle: true),
+      body: ListView(
+        padding: const EdgeInsets.symmetric(horizontal: 10, vertical: 10),
+        children: <Widget>[
+          ElevatedButton.icon(
+            onPressed: _scan,
+            icon: const Icon(Icons.camera),
+            label: const Text('Scan Product'),
+          ),
+          Card(
+            child: Column(
+              children: <Widget>[
+                if (product['name'] != null)
+                  ListTile(
+                    title: const Text('Name'),
+                    subtitle: Text(product['name']),
+                  ),
+                if (product['content'] != null)
+                  ListTile(
+                    title: const Text('Content'),
+                    subtitle: Text(product['content'].toString()),
+                  ),
+                if (product['category'] != null)
+                  ListTile(
+                    title: const Text('Category'),
+                    subtitle: Text(product['category']),
+                  ),
+                if (scanResult != null)
+                  ListTile(
+                    title: const Text('Barcode'),
+                    subtitle: Text(scanResult.rawContent),
+                  ),
+              ],
             ),
-          ],
-        ),
+          ),
+        ],
       ),
-      floatingActionButton: FloatingActionButton(
-        onPressed: _incrementCounter,
-        tooltip: 'Increment',
-        child: const Icon(Icons.add),
-      ), // This trailing comma makes auto-formatting nicer for build methods.
     );
   }
+
+  Future<void> fetchData() async {
+    final scanResult = this.scanResult;
+    try {
+      String? results = scanResult?.rawContent;
+      var url = Uri.parse('http://10.130.56.51/api/products/barcode/$results');
+      var response = await http.get(url);
+
+      if (response.statusCode == 200) {
+        var json = jsonDecode(response.body);
+
+        // If the response is data it is from the local database
+        if (json.containsKey("data")) {
+          setState(() {
+            product['name'] = jsonDecode(response.body)['data']['name'];
+            product['content'] = "${jsonDecode(response.body)['data']['content']} ${jsonDecode(response.body)['data']['unit']??''}";
+            product['category'] = jsonDecode(response.body)['data']['category'];
+          });
+        }
+
+        // If the response is instore it is from the backup database
+        if (json.containsKey("instore")) {
+          setState(() {
+            product['name'] = "${jsonDecode(response.body)['instore']['description']} ${jsonDecode(response.body)['instore']['name']}";
+            product['content'] = "${jsonDecode(response.body)['instore']['contents']} ${jsonDecode(response.body)['instore']['contentsUnit']}";            product['category'] = "None";
+          });
+        }
+
+        log(jsonEncode(product));
```

Comment:
Wild log spottet

Reply:
Logs not regarding error handling have been removed.